### ISSUE TITLE
Compilation fails on ubuntu with some flags

### ISSUE THREAD

**Issue by teto** [openntpd-portable/openntpd-portable#13]
hi,

I want to run openntpd inside a simulator thanks to the DCE project (https://www.nsnam.org/docs/dce/release/1.6/manual/html/dce-user-newapps.html). To achieve that, I need to compile openntpd with the following flags but it fails : any idea why please ?

```
$ CFLAGS="-fPIC -U_FORTIFY_SOURCE -g" LDFLAGS="-pie -rdynamic" ./configure                        
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/home/teto/openntpd-portable':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
```

Here is the config.log:
https://transfer.sh/T7UOo/config.log

**Comment by busterb** [comment]
Thanks, I see the issue. I need to modify the way user CFLAGS are handled during the configuration phase. Right now, configuration happens without user CFLAGS set, causing the linker flags to conflict in the earlier config phases.

**Comment by teto** [comment]
I would like to use openntpprotable in an experiment and this problem is blocking. Do you have an idea when you would be able to fix that ?

**Comment by busterb** [comment]
I just fixed the issue with 7523d1ea94ef9f35c60452cba377b67f52661b60, thanks for the report.

**Comment by teto** [comment]
I confirm it works. Thanks for the fix !
